shape.c

```diff
@@ -120,7 +348,7 @@ shape_alloc(void)
     shape_id_t shape_id = GET_SHAPE_TREE()->next_shape_id;
     GET_SHAPE_TREE()->next_shape_id++;
 
-    if (shape_id == MAX_SHAPE_ID) {
+    if (shape_id == (MAX_SHAPE_ID + 1)) {
```

Comment:
I wanted `MAX_SHAPE_ID` to be actually allocatable.  We could probably roll it back, but since `MAX_SHAPE_ID` wasn't actually allocatable the name seemed odd.